portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/BullhornServiceImpl.java

```diff
@@ -0,0 +1,339 @@
+                                WallItem wallItem = wallItemComment.getWallItem();
+                                String to = wallItem.getCreatorUuid();
+                                if (!to.equals(from)) {
+                                    Session session = sessionManager.getCurrentSession();
+                                    String oldUserId = session.getUserId();
+                                    session.setUserId(to);
+                                    String url = profileLinkLogic.getInternalDirectUrlToUserWall(wallItem.getUserUuid()
+                                                                                    , Long.toString(wallItem.getId()));
+                                    session.setUserId(oldUserId);
+                                    sqlService.dbInsert(null
+                                            , SOCIAL_INSERT_SQL
+                                            , new Object[] {from,to,event,ref,e.getEventTime(),url}
+                                            , "ID");
+                                    countCache.remove(to);
+                                }
+                            } catch (Exception exp) {
+                                log.error("Failed to store wall item alert.", exp);
+                            }
+                        } else if (ProfileConstants.EVENT_STATUS_UPDATE.equals(event)) {
+                            // Get all the posters friends
+                            Session session = sessionManager.getCurrentSession();
+                            String oldUserId = session.getUserId();
+                            session.setUserId(from);
+                            String url = profileLinkLogic.getInternalDirectUrlToUserProfile(from);
+                            List<Person> connections = profileConnectionsLogic.getConnectionsForUser(from);
+                            for (Person connection : connections) {
+                                String to = connection.getProfile().getUserUuid();
+                                sqlService.dbInsert(null
+                                        , SOCIAL_INSERT_SQL
+                                        , new Object[] {from,to,event,ref,e.getEventTime(),url}
+                                        , "ID");
+                                countCache.remove(to);
+                            }
+                            session.setUserId(oldUserId);
+                        } else if (ProfileConstants.EVENT_FRIEND_REQUEST.equals(event)) {
+                            String to = pathParts[2];
+                            Session session = sessionManager.getCurrentSession();
+                            String oldUserId = session.getUserId();
+                            session.setUserId(to);
+                            String url = profileLinkLogic.getInternalDirectUrlToUserConnections();
+                            session.setUserId(oldUserId);
```

Comment:
Seems like the above code is mostly repeated in every condition, so it can probably be moved outside to reduce code duplication.